src/legacy/core_plugins/expressions/public/np_ready/public/execute.ts

```diff
@@ -48,7 +48,7 @@ export class ExpressionDataHandler {
     }
 
     this.abortController = new AbortController();
-    this.inspectorAdapters = this.getActiveInspectorAdapters();
+    this.inspectorAdapters = params.inspectorAdapters || this.getActiveInspectorAdapters();
```

Comment:
if using loader we will want to reuse the same inspector adapters every time (we pass them in from loader) to make sure inspector updates when new data is requested